### ISSUE TITLE
Missing escaping of ? in note titles in manifest.xml for tomboy-ng 0.23

### ISSUE THREAD

**Issue by erik-brangs** [tomboy-notes/tomboy-ng#99]
I'm using tomboy-ng 0.23 on Windows. I had a note with a title like ``Foo "Text"?``. Then I deleted it. In ``manifest.xml``, this leads to an entry like

```
<note-deletions>
  <note guid="73A8C215-B3F7-4331-B449-EE8C36A9001E" title="Foo "Text"?" />
</note-deletions>
```
. When trying to delete another note, this leads to a crash with a message about expecting whitespace at a certain position.

**Comment by davidbannon** [comment]
Hmm, that sounds like a bad miss by me. Those double inverted commas definitely should not make it into the xml.  

As a stop gap, I suggest you delete the offending note on each machine and edit their manifest files.   I'll look through the code and it should be trivial to fix but won't me in a position to do a full test and release cycle for a couple of months. I'm traveling and don't have my Mac in particular nor my build and test machine.

Thanks for spotting this Erik, its a nasty one.

Davo

**Comment by davidbannon** [comment]
Yep, the issue, I think, is more about the double quote than the question mark.
In the Tomboy Note XML, " and ' do not need to be escaped as the note data is not presented as an attribute.  And I am being quite careful to maintain compatibility with Tomboy. However, the manifest files are not in TB note format, its general XML.  I am not sure how TB would react to a note named like that, I don't have my TB install with me and won't be home for several months.
But I am using (in tomboy-ng) std  XML tools to read the manifest files and they are going to choke on finding more than two double quotes in an attribute.
I am not sure how to handle this, I don't want to extend line number 163 of syncutils -
function RemoveBadXMLCharacters(const InStr : ANSIString) : ANSIString; 
to also escape ' and " because that may (or may not ? ) introduce a Tomboy incompatibility into our notes. But I must stop them getting through into the manifest file where it ends up as an attribute. 
Further research is indicated .......

**Comment by davidbannon** [comment]
Erik, I have committed a fix for this (as mentioned above). But won't be releasing a version for some months. Do you want me to cut you a Linux64 binary of were we are now ?
It will be v0.23 plus your fix (well tested) and a not so well tested set of changes trying to get a dark theme for windows that "should" have no impact on Linux at all. And a couple of unnecessary debug messages silenced...

Davo

**Comment by erik-brangs** [comment]
Removing the title attribute from the XML manifest file works for me as a workaround for now. I'll wait for the next regular release to get the bug fix.

Thank you.

**Comment by davidbannon** [comment]
Fixed in v0.24, thanks for report.
